prebid-server/developers/add-a-module-go.md

```diff
@@ -0,0 +1,266 @@
+2. `deps moduledeps.ModuleDeps` - contains dependencies that your module might require.
+
+and returns 2 values:
+1. `interface{}` - must implement at least 1 hook interface, see [hooks](#hook-interfaces). PBS uses type assertion to find out which hook interfaces implemented by module.
+2. `error` - any error occurred during module initialization.
+
+### Expose your module to PBS
+
+All available modules are exposed through the `modules/builder.go` file. This file is auto-generated, so you shouldn’t edit it manually.
+
+To register a new module, you just need to run one of the following commands from the PBS root directory:
+ - `make build-modules`
+ - or `go generate modules/modules.go`
+
+This command scans the `modules/` directory for files matching the pattern `modules/*/*/module.go` and adds all matching packages to the `modules/builder.go` file.
+
+## Module Code
+
+The quick start is to take a look in two places:
+- the [prebid ortb2blocking module](https://github.com/prebid/prebid-server/tree/master/modules/prebid/ortb2blocking)
+- the [hook source code and tests](https://github.com/prebid/prebid-server/tree/master/hooks)
+
+### Adding module documentation
+It is required to add a "README.md" file to the root of your module folder. It's recommended to specify the description of what the implemented module does, links to external documentation and include maintainer contact info (email, slack, etc).
+
+The documentation must also live on the docs.prebid.org site. Please add a markdown file to https://github.com/prebid/prebid.github.io/tree/master/prebid-server/pbs-modules
+
+### Hook Interfaces
+
+The Prebid server processing workflow is divided into several 'stages' where module authors can inject a specific function signature called a 'hook'.
+
+The Prebid Server host company will define which modules to run in which order by setting up a configuration defining which hooks run serially and which can run in parallel.
+
+The supported stages are described in the [general module overview](/prebid-server/developers/add-a-module.html#2-understand-the-endpoints-and-stages) and in PBS-Core source code at the "github.com/prebid/prebid-server/hooks" package.
+
+These are the available hooks that can be implemented in a module:
+
+- github.com/prebid/prebid-server/hooks/hookstage.Entrypoint
+- github.com/prebid/prebid-server/hooks/hookstage.RawAuctionRequest
+- github.com/prebid/prebid-server/hooks/hookstage.ProcessedAuctionRequest
+- github.com/prebid/prebid-server/hooks/hookstage.BidderRequest
+- github.com/prebid/prebid-server/hooks/hookstage.RawBidderResponse
+- github.com/prebid/prebid-server/hooks/hookstage.AllProcessedBidResponses
+- github.com/prebid/prebid-server/hooks/hookstage.AuctionResponse
+
+In a module it is not necessary to implement all mentioned interfaces but at least one is required by your functionality.
+
+### Examples
+
+1) To **update** the request in the `BidderRequest` hook you would return:
```

Comment:
Perhaps change this wording to something like this since the wording here only makes sense in the PBS-Java readme with their example just showing a promise that is returned:
"To **update** the request in the `BidderRequest` hook, your implementation would return a hook result with a change set:"